specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json

```
@@ -0,0 +1,179 @@
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+  	"/{scope}/providers/Microsoft.CostManagement/insights": {
+      "get": {
+        "tags": [
+          "Insights"
+        ],
+        "operationId": "Insights_Get",
+        "description": "Get Insights for the given scope",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/costmanagement/"
+        },
+        "x-ms-examples": {
+          "PrivateView": {
+            "$ref": "./examples/GetInsights.json"
+          }
+        },
+        "parameters": [
+		      {
+            "$ref": "#/parameters/scopeGetInsightsParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/InsightsCollectionResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+	  }
+  },
+  "definitions": {
+    "InsightsCollectionResponse": {
+      "description": "Result of listing Insights. It contains a list of available insight values.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of Insight values.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/Insights"
+          }
+        },
+        "nextLink": {
+          "description": "The link (url) to the next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "Insights": {
+      "description": "States and configurations of Cost Analysis.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ResourceModelWithAllowedPropertySet"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/InsightsProperties",
+          "title": "Insights properties"
+        }
+      }
+    },
+    "InsightsProperties": {
+      "type": "object",
+      "description": "The properties of the Insights.",
+      "properties": {
+        "status": {
+          "description": "Insight status kept from storage.",
```

Can we include available values? I believe this should be defined/controlled by the API, right?

```suggestion
          "description": "Status of the insight.",
```